### PR TITLE
feat(core-flows,medusa,pricing,types,utils): added price list workflows + endpoints

### PR DESCRIPTION
**Pull request by riqwan**
Apologies for the giant PR in advance, but most of these are removing of files and migrations from old workflows and routes to new.

What:

- Adds CRUD endpoints for price lists
- Migrate workflows from old sdk to new
- Added missing updatePriceListPrices method to pricing module

### CODE REVIEW

File: packages/core-flows/src/price-list/steps/delete-price-lists.ts
Comment (olivermrbl) [inline packages/core-flows/src/price-list/steps/delete-price-lists.ts:28]: question: Is there anything to implement? If you add it to the interface, you should be able to use it here, and the abstract module service will take care of the rest.

File: packages/types/src/pricing/common/price-list.ts
Comment (olivermrbl) [inline packages/types/src/pricing/common/price-list.ts]: todo: snake case
```suggestion
  price_list_id: string
```

File: packages/medusa/src/api-v2/admin/price-lists/route.ts
Comment (olivermrbl) [inline packages/medusa/src/api-v2/admin/price-lists/route.ts:52]: suggestion(non-blocking): Use `remoteQuery`

File: packages/medusa/src/api-v2/admin/price-lists/[id]/route.ts
Comment (olivermrbl) [inline packages/medusa/src/api-v2/admin/price-lists/[id]/route.ts]: todo: Should be idempotent, just return:
```
{
    id,
    object: "price_list",
    deleted: true,
  }
```

...even though the ID didn't exist.

File: packages/medusa/src/api-v2/admin/price-lists/[id]/route.ts
Comment (olivermrbl) [inline packages/medusa/src/api-v2/admin/price-lists/[id]/route.ts:59]: suggestion(non-blocking): Use `remoteQuery`

File: packages/medusa/src/api-v2/admin/price-lists/[id]/route.ts
Comment (olivermrbl) [inline packages/medusa/src/api-v2/admin/price-lists/[id]/route.ts]: thought(non-blocking): Would it make sense to move these to the workflow level?

What happens if the price list does not exist in the workflow? Do we handle that gracefully?

File: packages/core-flows/src/price-list/workflows/update-price-lists.ts
Comment (olivermrbl) [inline packages/core-flows/src/price-list/workflows/update-price-lists.ts]: todo: snake case
```suggestion
type WorkflowInput = { price_list_data: UpdatePriceListWorkflowInputDTO[] }
```

File: packages/core-flows/src/price-list/workflows/create-price-lists.ts
Comment (olivermrbl) [inline packages/core-flows/src/price-list/workflows/create-price-lists.ts]: todo: snake case

File: packages/core-flows/src/price-list/steps/delete-price-lists.ts
Comment (riqwan) [reply packages/core-flows/src/price-list/steps/delete-price-lists.ts:28]: Yes, unfortunately the models within aren't soft deletable-ready. Some of them don't even have the deleted_at attribute

File: packages/medusa/src/api-v2/admin/price-lists/route.ts
Comment (riqwan) [reply packages/medusa/src/api-v2/admin/price-lists/route.ts:52]: This function is already calling remote query inside

Comment (olivermrbl) [reply packages/medusa/src/api-v2/admin/price-lists/route.ts:52]: Ah my bad. Discard this and the other comments then.

**Comment by riqwan** [comment]
@olivermrbl addressed all your comments, shall we merge this in if it looks good?

File: integration-tests/modules/__tests__/price-lists/admin/price-lists.spec.ts
Comment (olivermrbl) [inline integration-tests/modules/__tests__/price-lists/admin/price-lists.spec.ts]: nit:
```suggestion
        it("should throw error when required attributes are not provided", async () => {
```

File: integration-tests/modules/__tests__/price-lists/admin/price-lists.spec.ts
Comment (olivermrbl) [inline integration-tests/modules/__tests__/price-lists/admin/price-lists.spec.ts:550]: suggestion: Assert on price ID

File: packages/medusa/src/api-v2/admin/price-lists/[id]/prices/route.ts
Comment (olivermrbl) [inline packages/medusa/src/api-v2/admin/price-lists/[id]/prices/route.ts:19]: suggestion: Should we move this to `POST /admin/price-lists/:id/prices/update`?

File: packages/medusa/src/api-v2/admin/price-lists/[id]/prices/route.ts
Comment (olivermrbl) [inline packages/medusa/src/api-v2/admin/price-lists/[id]/prices/route.ts:47]: suggestion: Should we move this to `POST /admin/price-lists/:id/prices/remove`?

File: packages/medusa/src/api-v2/admin/price-lists/middlewares.ts
Comment (olivermrbl) [inline packages/medusa/src/api-v2/admin/price-lists/middlewares.ts:59]: See other comment on endpoint paths

**Review by olivermrbl — APPROVED** [review]
LGTM, just a couple of things we need to align on before we merge

File: packages/medusa/src/api-v2/admin/price-lists/[id]/prices/route.ts
Comment (riqwan) [reply packages/medusa/src/api-v2/admin/price-lists/[id]/prices/route.ts:47]: Yes, it's already part of the work I'm doing next along with update prices.

**Comment by riqwan** [comment]
@olivermrbl will bring over the route changes to the next PR. 👍🏻 will stick to smaller PRs from here onward 😄

**Comment by olivermrbl** [comment]
sg!

**Pull request merged**
